Makefile.am

```diff
@@ -792,6 +792,7 @@ libneighbor_a_SOURCES += \
 NEIGHBOR_LIBS += \
 	$(EXPAT_LIBS) \
 	$(UPNP_LIBS)
+libneighbor_a_CXXFLAGS = $(UPNP_CFLAGS)
```

Comment:
This must be `_CPPFLAGS` because `UPNP_CFLAGS` contains preprocessor flags, not compiler flags.

Reply:
The output of `pkg-config --cflags ...` contains both pre-processor and compile flags and so both `CPPFLAGS` and `CXXFLAGS` are not optimal but IMHO `CFLAGS` is less wrong. Also the naming of the pkg-config stuff (`--cflags`, `PKG_CHECK_MODULES` setting `[$1]_CFLAGS` and not `[$1]_CPPFLAGS`) is a hint that CXXFLAGS is the better. In my universe using CFLAGS is also the more common.

But I'm not religious about that and if this is the only concern, I can rework.

Reply:
Oh, right, libupnp abuses this to inject `-pthread` into the using application...

Reply:
Note that `-pthread` is not a compile-time option, but a link-time option. It is wrong to add it to the `--cflags` output. This is a libupnp bug!

Reply:
I don't know if this is to be considered abuse or not, but looking at 
https://codesearch.debian.net/search?q=Cflags%3A.*-pthread&perpkg=1 and https://codesearch.debian.net/search?q=Libs%3A.*-pthread&perpkg=1 it doesn't look unusual at least. I will ask pkg-config upstream about the two things (-pthread handling and `CFLAGS` vs. `CPPFLAGS`)

Reply:
You don't need to ask pkg-config upstream, because this is not a pkg-config issue. And this is not a question about "CFLAGS vs CPPFLAGS" - because as I said, `-pthread` is not a compiler option at all! If anything, it belongs to `pkg-config --libs`.

Reply:
Clarification to avoid misunderstandings (because reality is more complex than my simplified explanation): yes, `-pthread` be can be useful at compile time and GCC documentation explains this, but **only** if the application to be compiled uses the thread API. For a library like libupnp, this is not applicable. libupnp itself should be compiled with `-pthread`, yes, but applications linking with libupnp do not need that flag.
Note that libupnp does ship with some header files which include `pthread.h` (of which MPD uses none), but these look like internal headers which are not part of the public API. I guess installing these as public headers was another mistake of the libupnp project.

Reply:
according to gcc docs `-pthread` is a preprocessor option and a linker option. Quoting `info gcc-7`:
> '-pthread'
>     Define additional macros required for using the POSIX threads
>     library.  You should use this option consistently for both
>     compilation and linking.

Reply:
I knew you would cite this, that's why I posted the clarification (at the same time).

Reply:
thanks for your insights, so for libupnp.pc the right thing to use is:
> Libs: -L${libdir} -lupnp -lthreadutil -lixml -pthread
> Cflags: -I${includedir}/upnp
(i.e. drop `-pthread` from `Cflags` and (slightly unrelated) move it to the end of `Libs` to get it into the right link order), right?